### PR TITLE
Virtual display feature added for linux os

### PR DESCRIPTION
**Pull request by dmytrokoren**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
 
-RUN apk add --update --no-cache chromium chromium-chromedriver
+RUN apk add --update --no-cache chromium chromium-chromedriver xvfb xauth
 
 RUN adduser -D auto-southwest-check-in -h /app
+RUN chown -R auto-southwest-check-in:auto-southwest-check-in /app
 USER auto-southwest-check-in
 
-COPY requirements.txt requirements.txt
+COPY requirements.txt ./
```
Comment (jdholtz) [inline Dockerfile:15]: What's the reason for this change? AFAIK, this change doesn't have any effect.

**Review by jdholtz — APPROVED** [review]
Looks good, thanks!

File: Dockerfile (hunk 1 above)
Comment (dmytrokoren) [reply Dockerfile:15]: The ./ notation is often used to indicate a relative path, which aligns with many Docker best practices, especially when you want to make it clear that you're copying to the current directory.

File: lib/webdriver.py [hunk 2]
```diff
@@ -273,3 +280,23 @@ def _set_account_name(self, account_monitor: AccountMonitor, response: JSON) ->
...
             f"{account_monitor.last_name}'s account\n"
         )  # Don't log as it contains sensitive information
+
+    def _quit_driver(self, driver: Driver) -> None:
+        driver.quit()
+        self._stop_display()
+
+    def _start_display(self) -> None:
+        try:
+            self.display = Display(visible=0, size=(1440, 1880))
```
Comment (jdholtz) [inline lib/webdriver.py:290]: It seems like `visible` is supposed to be a boolean and defaults to False. I'll remove it from being specified here. https://github.com/mdmintz/sbVirtualDisplay/blob/a6c725ba4e5d9cae1481bcea5bb7c708fc281a6f/sbvirtualdisplay/display.py#L15

File: Dockerfile [hunk 3]
```diff
@@ -1,17 +1,18 @@
-FROM python:3.12-rc-alpine
+FROM python:3.12-alpine3.17
```
Comment (jdholtz) [inline Dockerfile:1]: With this change, I am getting 403s during fare checks. However, I reverted it back to `rc-alpine` and have not reached any 403s. Can you confirm this @dmytrokoren?

Comment (dmytrokoren) [reply Dockerfile:1]: I have been running for 5 days without 403 on the 3.12-alpine3.17

After 2 intervals 403 error goes away... but let's keep 3.12-rc-alpine, I'm okay with it.

File: lib/webdriver.py (hunk 2 above)
Comment (dmytrokoren) [reply lib/webdriver.py]: Should we use False for (Xvfb)

Comment (jdholtz) [reply lib/webdriver.py]: Actually, I will add `backend = 'xvfb'` to ensure we always use XVFB

File: Dockerfile (hunk 3 above)
Comment (jdholtz) [reply Dockerfile:1]: Interesting... I reached a few 403s right away (enough to make the fare check unsuccessful, but none with `rc-alpine`.
